src/utils/api/btc/transactions.js

```diff
@@ -0,0 +1,232 @@
+    id: tx.txid,
+    timestamp: Number(timestamp) * 1000,
+    confirmations,
+    type: 0,
+    data: '',
+  };
+
+  data.fee = feeSatoshi;
+
+  const ownedInput = tx.inputs.find(i => i.txDetail.scriptPubKey.addresses.includes(address));
+
+  if (ownedInput) {
+    data.senderAddress = address;
+    const extractedAddress = tx.outputs[0].scriptPubKey.addresses[0];
+    data.recipientAddress = validateAddress(tokenMap.BTC.key, extractedAddress, netCode) === 0 ? extractedAddress : 'Unparsed Address';
+    data.amount = tx.outputs[0].satoshi;
+  } else {
+    const output = tx.outputs.find(o => o.scriptPubKey.addresses.includes(address));
+    const extractedAddress = tx.inputs[0].txDetail.scriptPubKey.addresses[0];
+    data.senderAddress = validateAddress(tokenMap.BTC.key, extractedAddress, netCode) === 0 ? extractedAddress : 'Unparsed Address';
+    data.recipientAddress = address;
+    data.amount = output.satoshi;
+  }
+
+  return data;
+});
+
+export const get = ({
+  id,
+  address,
+  limit = 50,
+  offset = 0,
+// eslint-disable-next-line max-statements
+}, netCode = 1) => new Promise(async (resolve, reject) => {
+  try {
+    let response;
+    const config = getBtcConfig(netCode);
+
+    if (id) {
+      response = await popsicle.get(`${config.url}/transaction/${id}`)
+        .use(popsicle.plugins.parse('json'));
+    } else {
+      response = await popsicle.get(`${config.url}/transactions/${address}?limit=${limit}&offset=${offset}&sort=height:desc`)
+        .use(popsicle.plugins.parse('json'));
+    }
+
+    const json = response.body;
+
+    if (response) {
+      const data = normalizeTransactionsResponse({
+        address,
+        list: id ? [json.data] : json.data,
+      });
+
+
+      resolve({
+        data,
+        meta: json.meta || {},
+      });
+    } else {
+      reject(json);
+    }
+  } catch (error) {
+    reject(error);
+  }
+});
+
+/**
+ * Normalizes transaction data retrieved from Blockchain.info API
+ * @param {Object} data
+ * @param {Number} data.inputCount
+ * @param {Number} data.outputCount
+ * @param {Number} data.dynamicFeePerByte - in satoshis/byte.
+ */
+export const calculateTransactionFee = ({
+  inputCount,
+  outputCount,
+  dynamicFeePerByte,
+}) => ((inputCount * 180) + (outputCount * 34) + 10 + inputCount) * dynamicFeePerByte;
+
+/**
+ * Retrieves unspent tx outputs of a BTC address from Blockchain.info API
+ * @param {String} address
+ * @returns {Promise<Array>}
+ */
+export const getUnspentTransactionOutputs = (address, netCode = 1) =>
+  new Promise(async (resolve, reject) => {
+    try {
+      const config = getBtcConfig(netCode);
+      const response = await popsicle.get(`${config.url}/utxo/${address}`)
+        .use(popsicle.plugins.parse('json'));
+      const json = response.body;
+
+      if (response) {
+        resolve(json.data);
+      } else {
+        reject(json);
+      }
+    } catch (error) {
+      reject(error);
+    }
+  });
+
+export const create = ({
```

Comment:
I think this function could be split in more functions, there are a lot of code inside this function that do different things, so maybe those could be in a different place

Reply:
Yeah, I was discussing this with @aydieneue on Friday, but we are not really sure on how to split this function without turning this more complex or what kind of logic would make sense to not keep on this function and move to a separate function, since in the end all the logic are only used to create the transaction.